src/app/components/sidenav/sidenav.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Icon } from 'src/app/models/icon';
+
+enum Role{
+  User=0,
+  Admin=1
+}
+
+@Component({
+  selector: 'app-sidenav',
+  templateUrl: './sidenav.component.html',
+  styleUrls: ['./sidenav.component.scss']
+})
+
+export class SidenavComponent implements OnInit {
+
+  @Input() isDescription = false;
+  
+  eRole=Role;
+  User=Role.Admin;
```

Comment:
Rename `User `to `userRole` - variables should be camel case - https://angular.io/guide/styleguide